### DISCUSSION TITLE
Default permissions granted to the GITHUB_TOKEN not respected

### DISCUSSION THREAD

**Discussion by riccardoporreca** [community/community#40423] [Actions] [answered]
### Select Topic Area

Bug

### Body

I have configured the GITHUB_TOKEN permissions to "Read and write permissions" in the Actions repository settings (https://github.com/riccardoporreca/rmdgallery).

![image](https://user-images.githubusercontent.com/13663564/204743890-90c66f1e-7a0c-457e-a641-109b416ec73c.png)

However, when I ran a workflow on the `main` branch, the logged permissions are read-only, see e.g. https://github.com/riccardoporreca/rmdgallery/actions/runs/3579429744/jobs/6024617600#step:1:16

```
GITHUB_TOKEN Permissions
  Actions: read
  Checks: read
  Contents: read
  Deployments: read
  Discussions: read
  Issues: read
  Metadata: read
  Packages: read
  Pages: read
  PullRequests: read
  RepositoryProjects: read
  SecurityEvents: read
  Statuses: read
```

I did try changing the settings to "Read repository contents permission" to confirm the settings are picked up, and indeed this reduced the logged permissions to (https://github.com/riccardoporreca/rmdgallery/actions/runs/3579429744/jobs/6020764515#step:1:16)

```
GITHUB_TOKEN Permissions
  Contents: read
  Metadata: read
```

I know I can (and I will) use explicit `permissions` in the workflow YAML to control this, and I understand it might be intentional to only allow read permissions for all scopes by default through the repository settings (thus forcing the usage of explicit `permissions` in the YAML), but there is currently an inconsistency between what the user sets in the settings and what is being used in the workflow.

Note that this was not an issue just few weeks ago https://github.com/riccardoporreca/rmdgallery/actions/runs/3448894356/jobs/5756333457#step:1:16

```
Sat, 12 Nov 2022 00:48:15 GMT GITHUB_TOKEN Permissions
Sat, 12 Nov 2022 00:48:15 GMT   Actions: write
Sat, 12 Nov 2022 00:48:15 GMT   Checks: write
Sat, 12 Nov 2022 00:48:15 GMT   Contents: write
Sat, 12 Nov 2022 00:48:15 GMT   Deployments: write
Sat, 12 Nov 2022 00:48:15 GMT   Discussions: write
Sat, 12 Nov 2022 00:48:15 GMT   Issues: write
Sat, 12 Nov 2022 00:48:15 GMT   Metadata: read
Sat, 12 Nov 2022 00:48:15 GMT   Packages: write
Sat, 12 Nov 2022 00:48:15 GMT   Pages: write
Sat, 12 Nov 2022 00:48:15 GMT   PullRequests: write
Sat, 12 Nov 2022 00:48:15 GMT   RepositoryProjects: write
Sat, 12 Nov 2022 00:48:15 GMT   SecurityEvents: write
Sat, 12 Nov 2022 00:48:15 GMT   Statuses: write
```

**Comment by riccardoporreca** [comment]
It looks like this is no issue when running

- on the same commit but from a different branch: https://github.com/community/community/discussions/40423#discussion-4614440
- on a new direct commit to `main` (no branch protection): https://github.com/riccardoporreca/rmdgallery/actions/runs/3581807020/jobs/6025296189#step:1:16
- on a new PR merge commit to `main` (no branch protection): https://github.com/riccardoporreca/rmdgallery/actions/runs/3581900377/jobs/6025495089#step:1:16
- on a new PR merge commit to `main` (with branch protection re-enabled): https://github.com/riccardoporreca/rmdgallery/actions/runs/3581963931/jobs/6025633220#step:1:16

**Comment by riccardoporreca** [comment, marked as answer]
I think I understand what is going on: The permission of `GITHUB_TOKEN` probably depend on the user triggering the ~pipeline~ workflow, which was **dependabot** for the dependency maintenance PR merged via `@dependabot merge`.

**Reply by riccardoporreca** [reply]
Here we go: https://docs.github.com/en/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions#responding-to-events

> For workflows initiated by Dependabot (github.actor == 'dependabot[bot]') using the pull_request, pull_request_review, pull_request_review_comment, push, create, deployment, and deployment_status events, the following restrictions apply:
> 
> - GITHUB_TOKEN has read-only permissions by default.
> - Secrets are populated from Dependabot secrets. GitHub Actions secrets are not available.

> By default, GitHub Actions workflows triggered by Dependabot get a GITHUB_TOKEN with read-only permissions. You can use the permissions key in your workflow to increase the access for the token:

This explain the issue and makes totally sense, so I will indeed rely on explicit (and minimal) `permissions` in the workflow YAML file

**Reply by Famousx39** [reply]
Sat, 12 Nov 2022 00:48:15 GMT GITHUB_TOKEN Permissions
Sat, 12 Nov 2022 00:48:15 GMT   Actions: write
Sat, 12 Nov 2022 00:48:15 GMT   Checks: write
Sat, 12 Nov 2022 00:48:15 GMT   Contents: write
Sat, 12 Nov 2022 00:48:15 GMT   Deployments: write
Sat, 12 Nov 2022 00:48:15 GMT   Discussions: write
Sat, 12 Nov 2022 00:48:15 GMT   Issues: write
Sat, 12 Nov 2022 00:48:15 GMT   Metadata: read
Sat, 12 Nov 2022 00:48:15 GMT   Packages: write
Sat, 12 Nov 2022 00:48:15 GMT   Pages: write
Sat, 12 Nov 2022 00:48:15 GMT   PullRequests: write
Sat, 12 Nov 2022 00:48:15 GMT   RepositoryProjects: write
Sat, 12 Nov 2022 00:48:15 GMT   SecurityEvents: write
Sat, 12 Nov 2022 00:48:15 GMT   Statuses: write
